src/IW/Effects/Cabal.hs

```diff
@@ -0,0 +1,61 @@
+{- | This module contains the class definitiion of @MonadCabal@ and
+an instance of @MonadCabald@ for the @App@ monad. Instances of
+@MonadCabal@ have a @getCabalCategories@ action that returns @[Category]@
+given a @RepoOwner@ and @RepoName@.
+-}
+
+module IW.Effects.Cabal
+       ( getCabalCategories
+
+       -- * Internals
+       , getCabalCategoriesImpl
+       , repoCabalUrl
+       ) where
+
+import Data.Text (splitOn, strip)
+import Distribution.PackageDescription
+import Distribution.PackageDescription.Parsec (parseGenericPackageDescriptionMaybe)
+
+import IW.App (App)
+import IW.Core.Repo (RepoOwner (..), RepoName (..), Category (..))
+import IW.Core.Url (Url (..))
+import IW.Effects.Download (MonadDownload (..))
+
+
+-- | Describes a monad that returns @[Category]@ given a @RepoOwner@ and @RepoName@.
+class Monad m => MonadCabal m where
+    getCabalCategories :: RepoOwner -> RepoName -> m [Category]
+
+instance MonadCabal App where
+    getCabalCategories = getCabalCategoriesImpl
+
+getCabalCategoriesImpl :: MonadDownload m => RepoOwner -> RepoName -> m [Category]
+getCabalCategoriesImpl repoOwner repoName = do
+    cabalFile <- downloadFile $ repoCabalUrl repoOwner repoName
+    pure $ case parseGenericPackageDescriptionMaybe cabalFile of
+        Nothing -> []
```

Comment:
Could we add log that parsing failed? This is an interesting case for us to look into in the future.

Reply:
@vrom911 Added.